array-reduce/main.js

```diff
@@ -0,0 +1,41 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+// [1, 2, 3, 4];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((previousValue, currentValue) => previousValue + currentValue);
+console.log('sum:', sum);
+
+const product = numbers.reduce((previousValue, currentValue) => previousValue * currentValue);
+console.log('product:', product);
+
+let totalBalance = 0;
+const balance = account.reduce((previousValue, currentValue) => {
+  if (currentValue.type === 'deposit') {
+    totalBalance = previousValue + currentValue.amount;
+  } else {
+    totalBalance = previousValue - currentValue.amount;
+  }
+  return totalBalance;
```

Comment:
Inside the callback you want to be using `previousValue` everywhere you used `totalBalance`, you don't want to rely on values outside the function.